libmscore/rest.cpp

```diff
@@ -90,29 +102,32 @@ void Rest::draw(QPainter* painter) const
             //only on voice 1
             if (track() % VOICES)
                   return;
-            Measure* m = measure();
-            int n      = m->mmRestCount();
+
+            // draw number
+            int n      = measure()->mmRestCount();
+            std::vector<SymId>&& s = toTimeSigString(QString("%1").arg(n));
+            qreal y = _yPosMMRestNumber * spatium() - staff()->height() * .5;
+            qreal x = _mmWidth * .5;
+            x      -= symBbox(s).width() * .5;
+            drawSymbols(s, painter, QPointF(x, y));
+
+            // draw horizontal line
             qreal pw   = _spatium * .7;
             QPen pen(painter->pen());
             pen.setWidthF(pw);
             painter->setPen(pen);
-
-            qreal w  = _mmWidth;
-            qreal x2 =  w;
-            pw *= .5;
-            painter->drawLine(QLineF(pw, 0.0, _mmWidth - pw, 0.0));
-
-            // draw vertical lines:
+            qreal x1 = pw * .5;
+            qreal x2 = _mmWidth - x1;
+//            if (_yPosMMRestNumber > 0.7) // hack: "magic number" for when it encounters horizontal line
```

Comment:
what is the purpose of this comment?

Reply:
It's a reminder to myself about another feature I'd like to add, which would basically automatically detect when the number would collide with the horizontal bar and and separate the bar into two segments with a space big enough for the number. I've removed the comment for now.